backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/query/impl/LuceneSearchScrollImpl.java

```diff
@@ -93,7 +98,12 @@ public void close() {
 
 		// no more results check
 		if ( scrollIndex >= search.hitSize() ) {
-			return new LuceneSearchScrollResultImpl<>( false, Collections.emptyList(),
+			if ( cachedTotal == null ) {
+				cachedTotal = ( timeoutManager.isTimedOut() ) ? SimpleSearchResultTotal.lowerBound( 0L ) :
+						SimpleSearchResultTotal.exact( 0L );
```

Comment:
That would be `search.hitSize()`, not `0`. Same above.

That being said... you could expose the `SearchResultTotal` in `LuceneExtractableSearchResult`, and use that. It's already there, accessible from `luceneCollectors.getResultTotal()`, and you know it's correct, so why not use it?

Going one step further, you could even rethink this method: if we're going beyond the end, in this block you would just set a boolean `hasHits = false`, and let this method continue to the last line, where you pass the boolean to the constructor of the search result... I think it may work as expected?

Reply:
All good ides... I'm trying to applying them.

Reply:
I applied the first two. About the the last one, the good fact of the current solution is that will be sure to avoid any `#loadBlocking` if we know in advance that we don't need it.

Reply:
> I applied the first two

Thanks

> About the the last one, the good fact of the current solution is that will be sure to avoid any #loadBlocking if we know in advance that we don't need it.

`loadBlocking` won't do anything if there is nothing to load, so that's not a very good reason. Anyway, it's not that important, so let's forget about it.